lib/ansible/plugins/inventory/aws_ec2.py

```diff
@@ -477,12 +477,13 @@ def _add_hosts(self, hosts, group, hostnames):
 
             # Composed variables
             self._set_composite_vars(self.get_option('compose'), host, hostname, strict=strict)
+            variables = self.inventory.get_host(hostname).vars
 
             # Complex groups based on jinja2 conditionals, hosts that meet the conditional are added to group
-            self._add_host_to_composed_groups(self.get_option('groups'), host, hostname, strict=strict)
+            self._add_host_to_composed_groups(self.get_option('groups'), variables, hostname, strict=strict)
```

Comment:
instead of changing the signature (3rd party plugins will break), why not add the get variables called inside the composed function?

Reply:
@bcoca Should composed variables win, or the variables that are provided to the method?

Reply:
last defined wins, so in this case composed should win over provided

Reply:
Okay, thanks. Did that.